### DISCUSSION TITLE
Benchmarks

### DISCUSSION THREAD

**Discussion by dmarro89** [dmarro89/dare-db#39] [General]
I'm opening this discussion to start thinking about some benchmarks.
Please feel free to contribute @vdmitriyev @TheInvincibleRalph .

**Comment by vdmitriyev** [comment]
It is a very good idea to have a solid benchmark to evaluated performance of a database.  As a very first step, a "plan of a benchmark" would be great to have. This plan should specify, what is going to be benchmarked. For instance, at least 2 different aspects should be considered for a benchmark: 

1. consistency of a data in a database under different loads
2. speed of response on writes/reads under different loads

Here are some general articles/references, which could be useful. 

* Tool, which could be used for a benchmark of HTTP
	+ HTTP load generator, ApacheBench (ab) replacement - https://github.com/rakyll/hey
* An article with a benchmark example
	+ CPU vs I/O Bound Benchmarking in Go - https://medium.com/@halilylm/cpu-vs-i-o-bound-benchmarking-in-go-fcd4f053694e

**Reply by dmarro89** [reply]
Thanks @vdmitriyev.
I'll raise a new Issue about the Benchmarks implementation.

**Comment by dmarro89** [comment]
Let's keep this discussion opened in order to collect more ideas about it.
